### PR TITLE
CAdES: rework CAdES signing API.

### PR DESCRIPTION
**Pull request by FdaSilvaYY**
Make it private, as it is unused outside library bounds.

Refer https://github.com/openssl/openssl/issues/9095#issuecomment-500249260

##### Checklist

- [X] documentation is added or updated
- [X] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
Have these API's appeared in any release (or are they only in master?)

**Comment by FdaSilvaYY** [comment]
They were added to master only, so it is not too late to cut it out.
I need to figure out the right way to do it, that's why it is a WIP.

**Comment by mattcaswell** [comment]
Travis failures look relevant.

File: crypto/cms/cms_lcl.h [hunk 1]
```diff
@@ -419,6 +419,10 @@ int cms_RecipientInfo_pwri_crypt(const CMS_ContentInfo *cms, CMS_RecipientInfo *
 /* SignerInfo routines */
 int CMS_si_check_attributes(const CMS_SignerInfo *si);
 
+/* ESS routines */
+int CMS_add1_signing_cert(CMS_SignerInfo *si, ESS_SIGNING_CERT *sc);
```
Comment (slontis) [inline crypto/cms/cms_lcl.h]: use lowercase cms now that they are local e.g. cms_add1_signing_cert

**Comment by FdaSilvaYY** [comment]
Waiting for a new CI build results, after some doc-nits cleanup.
Ready for review.

File: crypto/cms/cms_lcl.h (hunk 1 above)
Comment (FdaSilvaYY) [reply crypto/cms/cms_lcl.h]: done !

**Comment by FdaSilvaYY** [comment]
Ping @openssl : just needs a second (OMC?) review.

**Comment by opensignature** [comment]
Looks good to me. Thanks.

File: crypto/include/internal/ess_int.h [hunk 2]
```diff
@@ -12,9 +12,18 @@
 ESS_SIGNING_CERT *ESS_SIGNING_CERT_get(PKCS7_SIGNER_INFO *si);
 int ESS_SIGNING_CERT_add(PKCS7_SIGNER_INFO *si, ESS_SIGNING_CERT *sc);
 
+ESS_SIGNING_CERT *ESS_SIGNING_CERT_new_init(X509 *signcert,
+                                            STACK_OF(X509) *certs,
+                                            int issuer_needed);
```
Comment (mspncp) [inline crypto/include/internal/ess_int.h]: _Just a note, not a change request: This is an example of conflicting naming conventions: OOH, the function is internal so the prefix should be lowercase, OTOH, all accessors for a FOO 'class' are normally called FOO_method_name(), with the classname in uppercase._

**Comment by slontis** [comment]
ping @openssl/omc reviewers

**Comment by paulidale** [comment]
Merged to master.  Thanks all.

**Comment by FdaSilvaYY** [comment]
Commit 8c00f267b8df1a8c70eff8198de40aa561299e48

**Pull request closed without merging**